### ISSUE TITLE
Molmo-7B-D 4bit suddenly stopped working

### ISSUE THREAD

**Issue by mkagenius** [Blaizzy/mlx-vlm#160]
It was working yesterday (25/12/24) and now without any underlying change (I think) it starts complaining about missing params.

`ValueError: Missing parameters: vision_tower.image_vit.patch_embedding.biases vision_tower.image_vit.patch_embedding.scales.`

Full command - 
(even without image, it gives same error)

```
(myenv) (miniconda3) ~/clickclickclick$ python3.12 -m mlx_vlm.generate --model mlx-community/Molmo-7B-D-0924-4bit --max-tokens 100 --temp 0.0 --image http://images.cocodataset.org/val2017/000000039769.jpg
Fetching 15 files: 100%|███████████████████████████████████████████████████████████████████| 15/15 [00:00<00:00, 246723.76it/s]
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "/Users/manish/clickclickclick/myenv/lib/python3.12/site-packages/mlx_vlm/generate.py", line 111, in <module>
    main()
  File "/Users/manish/clickclickclick/myenv/lib/python3.12/site-packages/mlx_vlm/generate.py", line 80, in main
    model, processor, image_processor, config = get_model_and_processors(
                                                ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/manish/clickclickclick/myenv/lib/python3.12/site-packages/mlx_vlm/generate.py", line 68, in get_model_and_processors
    model, processor = load(
                       ^^^^^
  File "/Users/manish/clickclickclick/myenv/lib/python3.12/site-packages/mlx_vlm/utils.py", line 292, in load
    model = load_model(model_path, lazy)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/manish/clickclickclick/myenv/lib/python3.12/site-packages/mlx_vlm/utils.py", line 257, in load_model
    model.load_weights(list(weights.items()))
  File "/Users/manish/clickclickclick/myenv/lib/python3.12/site-packages/mlx/nn/layers/base.py", line 181, in load_weights
    raise ValueError(f"Missing parameters: {missing}.")
ValueError: Missing parameters: vision_tower.image_vit.patch_embedding.biases vision_tower.image_vit.patch_embedding.scales.
(myenv) (miniconda3) ~/clickclickclick$ 
(myenv) (miniconda3) ~/clickclickclick$ 
(myenv) (miniconda3) ~/clickclickclick$ pip3.12 freeze | grep torch
torch==2.5.1
torchvision==0.20.1
(myenv) (miniconda3) ~/clickclickclick$ pip3.12 freeze | grep mlx
mlx==0.21.1
mlx-vlm==0.1.6
(myenv) (miniconda3) ~/clickclickclick$ 
```

Its a little weird, since neither mlx, nor mlx-vlm has changed yesterday (nor the weights from mlx-community on huggingface)

**Comment by mkagenius** [comment]
[temp.txt](https://github.com/user-attachments/files/18255248/temp.txt)
All the keys. Those two keys indeed seem missing but maybe present with some other name, as some sanitization(transformation) was happening to couple of other keys too.

**Comment by mkagenius** [comment]
Oh, there was this change in the hugging-face 3 hours ago - https://huggingface.co/mlx-community/Molmo-7B-D-0924-4bit/commit/826ce5935256778523bfee774845f8387a2d6e3e

<img width="1567" alt="image" src="https://github.com/user-attachments/assets/b19f3126-3058-4515-a3b4-269cecf42420" />

And exactly those keys are missing.

**Comment by Blaizzy** [comment]
My bad!

I'm refactoring utile and because of it I had change the model configs. 

I will make a new release fixing it in a few.

**Comment by Blaizzy** [comment]
Bear with me :)

**Comment by Blaizzy** [comment]
You can use this branch for now 
https://github.com/Blaizzy/mlx-vlm/tree/pc/refactor-utils-1

**Comment by mkagenius** [comment]
I tried the branch but it has the same issue of missing keys, as the hugging face model's json itself is no longer the same.

No issues, I will wait for the refactor to finish.

**Comment by mkagenius** [comment]
> Oh, there was this change in the hugging-face 3 hours ago - https://huggingface.co/mlx-community/Molmo-7B-D-0924-4bit/commit/826ce5935256778523bfee774845f8387a2d6e3e
> 
> <img alt="image" width="1567" src="https://private-user-images.githubusercontent.com/1682656/398800836-b19f3126-3058-4515-a3b4-269cecf42420.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzUyNzg5MDksIm5iZiI6MTczNTI3ODYwOSwicGF0aCI6Ii8xNjgyNjU2LzM5ODgwMDgzNi1iMTlmMzEyNi0zMDU4LTQ1MTUtYTNiNC0yNjljZWNmNDI0MjAucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTIyNyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEyMjdUMDU1MDA5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MjlkM2JlYWIzZjIyMjhhNzk4MjkyODc2MDAyNmRhY2FhMjRiYzEzMDdmYzc2MzJkNTIwZmM0ZjgzM2FmMGZjYSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.HGRNv-_kWFDXCCkWE_k781iX3UlZo8hwQmzttsBEkwU">
> And exactly those keys are missing.

Seems the keys were removed a month ago, not 3 hours ago. 3 hours ago there was a minor change not related to keys:

<img width="1526" alt="image" src="https://github.com/user-attachments/assets/b049cf01-53b6-4faf-8da5-258392add09f" />

So, I am confused now what caused the issue then. I have tried 0.1.4 and 0.1.6 (and the refactor utils branch)

**Comment by Blaizzy** [comment]
It's fixed ✅

Please use the branch in this PR #161

**Comment by Blaizzy** [comment]
What caused the issue is a change in the config file that is necessary for the upcoming changes.

**Comment by mkagenius** [comment]
Works now. The released 0.1.6 itself worked (I havent tested the branch).

**Comment by Blaizzy** [comment]
Thanks! 

I found a better solution that is backwards compatible
